components/eamxx/src/share/tests/refining_remapper_tests.cpp

```diff
@@ -0,0 +1,467 @@
+#include <catch2/catch.hpp>
+
+#include "share/grid/remap/refining_remapper.hpp"
+#include "share/grid/point_grid.hpp"
+#include "share/io/scream_scorpio_interface.hpp"
+#include "share/util/scream_setup_random_test.hpp"
+#include "share/field/field_utils.hpp"
+
+namespace scream {
+
+template<typename VT>
+typename VT::HostMirror
+cmvdc (const VT& v) {
+  auto vh = Kokkos::create_mirror_view(v);
+  Kokkos::deep_copy(vh,v);
+  return vh;
+}
+
+class RefiningRemapperTester : public RefiningRemapper {
+public:
+  RefiningRemapperTester (const grid_ptr_type& tgt_grid,
+                          const std::string& map_file)
+   : RefiningRemapper(tgt_grid,map_file) {}
+
+  ~RefiningRemapperTester () = default;
+
+  void test_internals () {
+    // Test arrays size
+    const size_t n = m_num_fields;
+    REQUIRE (m_src_fields.size()==n);
+    REQUIRE (m_ov_src_fields.size()==n);
+    REQUIRE (m_tgt_fields.size()==n);
+    REQUIRE (m_col_size.size()==n);
+    REQUIRE (m_col_stride.size()==n);
+    REQUIRE (m_col_offset.size()==n);
+    REQUIRE (m_mpi_win.size()==n);
+    REQUIRE (m_remote_lids.size()==static_cast<size_t>(m_ov_src_grid->get_num_local_dofs()));
+    REQUIRE (m_remote_pids.size()==static_cast<size_t>(m_ov_src_grid->get_num_local_dofs()));
+
+    // Test field specs
+    constexpr auto COL = ShortFieldTagsNames::COL;
+    for (int i=0; i<m_num_fields; ++i) {
+      const auto& fh = m_src_fields[i].get_header();
+      REQUIRE (m_col_size[i]==fh.get_identifier().get_layout().strip_dim(COL).size());
+      if (fh.get_parent().lock()) {
+        REQUIRE (m_col_stride[i]==m_col_size[i]*fh.get_alloc_properties().get_subview_info().dim_extent);
+        REQUIRE (m_col_offset[i]==m_col_size[i]*fh.get_alloc_properties().get_subview_info().slice_idx);
+      } else {
+        REQUIRE (m_col_stride[i]==m_col_size[i]);
+        REQUIRE (m_col_offset[i]==0);
+      }
+    }
+
+    // Test CRS matirx
+    int nldofs_tgt = m_tgt_grid->get_num_local_dofs();
+    int ngdofs_src = m_src_grid->get_num_global_dofs();
+    REQUIRE (m_row_offsets.extent_int(0)==nldofs_tgt+1);
+    auto row_offsets_h = cmvdc(m_row_offsets);
+    auto col_lids_h = cmvdc(m_col_lids);
+    auto weights_h = cmvdc(m_weights);
+    auto col_gids_h = m_ov_src_grid->get_dofs_gids().get_view<const AbstractGrid::gid_type*,Host>();
+
+    auto row_gids_h = m_tgt_grid->get_dofs_gids().get_view<const AbstractGrid::gid_type*,Host>();
+    for (int i=0; i<nldofs_tgt; ++i) {
+      auto row_gid = row_gids_h[i];
+      auto beg = row_offsets_h(i);
+      auto end = row_offsets_h(i+1);
+      REQUIRE (end>=beg);
+      for (int j=beg; j<end; ++j) {
+        if (row_gid<ngdofs_src) {
+          REQUIRE (weights_h(j)==1);
+        } else {
+          REQUIRE (weights_h(j)==0.5);
+        }
+      }
+    }
+  }
+};
+
+Field create_field (const std::string& name, const LayoutType lt, const AbstractGrid& grid)
```

Comment:
None of these fields are packed.  Would it make sense just to include at least one packed field to have more coverage?